client/src/main/java/tds/session/ExternalSessionConfiguration.java

```diff
@@ -58,4 +61,22 @@ public int getShiftWindowStart() {
     public int getShiftWindowEnd() {
         return shiftWindowEnd;
     }
+
+    /**
+     * Determine if this {@link ExternalSessionConfiguration} is configured for the Simulation environment.
+     *
+     * @return True if the environment is set to "simulation" (case-insensitive); otherwise false.
+     */
+    public boolean isInSimulationEnvironment() {
+        return this.getEnvironment().equalsIgnoreCase(SIMULATION_ENVIRONMENT);
```

Comment:
I believe this will throw if `environment` is null so add a null check?  Here and below.